muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/chat/ChatRoomSyncScheduler.java

```diff
@@ -83,59 +106,94 @@ public void updateChatRooms() {
+        curriculumIdentifiers = new ArrayList<>(curriculums.size());
+        for (Curriculum curriculum : curriculums) {
+          SchoolDataIdentifier curriculumIdentifier = SchoolDataIdentifier.fromId(curriculum.getIdentifier().toString());
+          if (curriculumIdentifier != null && !"OPS2005".equals(curriculum.getName())) {
+            curriculumIdentifiers.add(curriculumIdentifier);
+          } 
+        }
+      }
+      
+      String searchString = null;
+      boolean includeUnpublished = true;
+	  int firstResult = 0;
+  	  int maxResults = 100;
+	  List<Sort> sorts = null;
+	  searchResult = searchProvider.searchWorkspaces(null, null, null, null, curriculumIdentifiers, searchString, null, null, includeUnpublished, firstResult, maxResults, sorts);          
+	  
+	  List<Map<String, Object>> results = searchResult.getResults();
+      for (Map<String, Object> result : results) {
+    	  String searchId = (String) result.get("id");
+          if (StringUtils.isNotBlank(searchId)) {
+            String[] id = searchId.split("/", 2);
+            if (id.length == 2) {
+              String dataSource = id[1];
+              String identifier = id[0];
+              WorkspaceEntity workspaceEntity = workspaceEntityController.findWorkspaceByDataSourceAndIdentifier(dataSource, identifier);
+            
+              String name = (String) result.get("name");
+                
+        	  try {
+        		// Checking before creating is subject to a race condition, but in the worst
+        	    // case
+                // the creation just fails, resulting in a log entry
+       	        MUCRoomEntity chatRoomEntity = client.getChatRoom(identifier);
+        		  if (chatRoomEntity == null) {
+        		    logger.log(Level.INFO, "Syncing chat workspace " + name);
+        		    // SchoolDataIdentifier identifier = SchoolDataIdentifier.fromId(enabledWorkspace.getIdentifier());
+        		    if (identifier == null) {
+        		      logger.log(Level.WARNING, "Invalid workspace identifier " + identifier + ", skipping...");
+        		      continue;
+        		    }
+        		  
+        		    if (workspaceEntity == null) {
+        		      logger.log(Level.WARNING, "No workspace entity found for identifier " + identifier + ", skipping...");
+        		      continue;
+        		    }
+        		    Workspace workspace = workspaceController.findWorkspace(workspaceEntity);
+                String subjectCode = courseMetaController.findSubject(workspace.getSchoolDataSource(), workspace.getSubjectIdentifier()).getCode();
+                
+        		      String roomName = subjectCode + workspace.getCourseNumber() + " - " + workspace.getNameExtension();
+        		      chatRoomEntity = new MUCRoomEntity(identifier, roomName, workspace.getDescription());
+        		      client.createChatRoom(chatRoomEntity);
+
+        		      List<WorkspaceUser> workspaceUsers = workspaceController.listWorkspaceStudents(workspaceEntity);
```

Comment:
Especially for large non-stop courses, retrieving and going through all of their students might put the servers under severe stress.

It would be better if chat room admins and members were maintained based on Muikku receiving events that people have been added to (or removed from) workspaces.